src/fivetran-destination/src/utils.rs

```diff
@@ -98,3 +103,262 @@ pub fn to_fivetran_type(ty: Type) -> Result<(DataType, Option<DecimalParams>), O
         }
     }
 }
+
+/// Maps the column ordering of a CSV file, to the ordering of the provided table.
+///
+/// The column ordering of a CSV provided by Fivetran is not guaranteed to match the column
+/// ordering of the destination table, so based on column name we re-map the CSV.
+#[derive(Debug)]
+pub struct AsyncCsvReaderTableAdapter<R> {
+    /// Async CSV reader.
+    csv_reader: csv_async::AsyncReader<R>,
+    /// The `i`th column in a table, exists at the `tbl_to_csv[i]` position in a CSV record.
+    tbl_to_csv: Vec<usize>,
+}
+
+impl<R> AsyncCsvReaderTableAdapter<R>
+where
+    R: AsyncRead + Unpin + Send + 'static,
+{
+    /// Creates a new [`AsyncCsvReaderTableAdapter`] from an [`AsyncRead`]-er and a [`Table`].
+    pub async fn new(reader: R, table: &Table) -> Result<Self, OpError> {
+        let mut csv_reader = csv_async::AsyncReaderBuilder::new().create_reader(reader);
+        let csv_headers = csv_reader.headers().await?;
+        let csv_columns: BTreeMap<_, _> = csv_headers
+            .iter()
+            .enumerate()
+            .map(|(idx, name)| (name.trim().to_string(), idx))
```

Comment:
Why does this need `trim()`ing? At least in postgres it is valid to have a column name be a single space character.

Reply:
Great point, when looking at some sample CSV data at some point I thought I saw erroneous white space, but you're totally right, we shouldn't be trimming here. I added a test case to assert some interesting scenarios with leading spaces.